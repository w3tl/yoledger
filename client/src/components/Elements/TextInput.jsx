import React from 'react';
import PropTypes from 'prop-types';
import { withLabel } from './utils';

function TextInput({ value, onChange, ...props }) {
  return (
    <input value={value} onChange={onChange} {...props} />
  );
}

TextInput.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func,
};

TextInput.defaultProps = {
  value: '',
  onChange: () => {},
};

export default withLabel(TextInput);
