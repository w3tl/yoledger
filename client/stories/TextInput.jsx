import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { TextInput } from '../src/components/Elements';

storiesOf('Elements/TextInput', module)
  .add('default', () => <TextInput />)
  .add('with label', () => <TextInput label="Input here" />)
  .add('with value', () => <TextInput value={10} onChange={action('Value changed')} />);
