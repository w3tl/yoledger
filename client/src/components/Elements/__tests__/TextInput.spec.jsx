import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import TextInput from '../TextInput';

configure({ adapter: new Adapter() });

describe('TextInput', () => {
  it('should render correctly', () => {
    const output = shallow(
      <TextInput value="text" />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should handle the input event', () => {
    const onChange = jest.fn();
    const output = shallow(
      <TextInput value="text" onChange={onChange} />,
    );
    output.simulate('change', 'new text');
    expect(onChange).toHaveBeenCalledWith('new text');
  });
});
