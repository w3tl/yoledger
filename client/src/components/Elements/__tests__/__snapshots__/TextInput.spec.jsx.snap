// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<TextInput
  id=""
  onChange={[Function]}
  value="text"
/>
`;
